### PR TITLE
streaming service in func and hashtag streaming service in warble

### PR DESCRIPTION
**Pull request by gabroo**

### CODE REVIEW

**Comment by camshannon** [comment]
For the sake of time, I think you should go ahead and merge this. I looked over it, and it makes sense. I did not see any obvious bugs.

File: build/cpp/warble/inc/warble_functions.h [hunk 1]
```diff
@@ -122,26 +117,31 @@ ReadReplyPackager(const reply_vector &result);
...
-         std::make_pair(ProfileRequestPackager, ProfileReplyPackager)}};
-} // namespace warble_functions
\ No newline at end of file
+const function_mapping func_map{
+  {"Registeruser",
+    std::make_pair(RegisteruserRequestPackager, RegisteruserReplyPackager)},
+  {"Warble", std::make_pair(WarbleRequestPackager, WarbleReplyPackager)},
+  {"Follow", std::make_pair(FollowRequestPackager, FollowReplyPackager)},
+  {"Read", std::make_pair(ReadRequestPackager, ReadReplyPackager)},
+  {"Profile", std::make_pair(ProfileRequestPackager, ProfileReplyPackager)}
```
Comment (camshannon) [inline build/cpp/warble/inc/warble_functions.h]: You may have intended this, but Stream is not in this func_map

File: build/cpp/cmd/src/cmd.cc [hunk 2]
```diff
@@ -11,12 +11,13 @@ cmd::CommandLine::~CommandLine() { delete func_client_; }
...
-  func_client_->hook(2, "Follow");
-  func_client_->hook(3, "Read");
-  func_client_->hook(4, "Profile");
-  std::cout << "Warble functions successfully hooked." << std::endl;
+  func_client_->Hook(0, "Registeruser");
+  func_client_->Hook(1, "Warble");
+  func_client_->Hook(2, "Follow");
+  func_client_->Hook(3, "Read");
+  func_client_->Hook(4, "Profile");
+  func_client_->Hook(1, "Hashtags");
```
Comment (camshannon) [inline build/cpp/cmd/src/cmd.cc]: One thing I noticed is you hook Hashtags with the value 1 along with Warble, but it looks like this was intentional.

Comment (gabroo) [reply build/cpp/cmd/src/cmd.cc]: Thanks for pointing out. It should have been deleted. The stream map used in func is loaded in when the server starts

File: build/cpp/warble/inc/warble_functions.h (hunk 1 above)
Comment (gabroo) [reply build/cpp/warble/inc/warble_functions.h]: Yes, there is a separate map used for stream events, defined in `warble_functions.h`

**Pull request merged**